cdap-api/src/main/java/co/cask/cdap/api/mapreduce/MapReduceContext.java

```diff
@@ -117,6 +117,25 @@
   void setOutput(String datasetName, Dataset dataset);
 
   /**
+   * Overrides the output configuration of this MapReduce job to also allow writing to the specified dataset by
+   * its name.
+   *
+   * @param datasetName the name of the output dataset
+   */
+  void addOutput(String datasetName);
+
+  /**
+   * Overrides the output configuration of this MapReduce job to also allow writing to the specified dataset instance.
+   * Currently, the dataset passed in must be an {@link OutputFormatProvider}.
+   * You may want to use this method instead of {@link #setOutput(String)} if your output dataset uses runtime
+   * arguments set in your own program logic.
+   *
+   * @param datasetName the name of the output dataset
+   * @param dataset the output dataset
+   */
+  void addOutput(String datasetName, Dataset dataset);
```

Comment:
Since this is a new method, we shouldn't take a `Dataset` instance, but a dataset argument instead (Map<String, String>)